### DISCUSSION TITLE
Action order question

### DISCUSSION THREAD

**Discussion by ravikirancopart** [statelyai/xstate#1845] [General]
I know we should not rely on the order of the actions, but is there ever a possibility of a **sendparent** action to be called before an **assign** action is completed. Can a race condition occur here?
`actions:[assign(()=>{....}), sendParent(()=>{.....})]`

**Comment by Andarist** [comment]
In the current v4 assign will always be called first, even for `[sendParent(/* ... */, assign(/* ... */)]`. However, in the next major version actions will always be called in the definition order.

**Reply by ravikirancopart** [reply]
Thanks for that bit of info @Andarist . The reason I'm asking this is, we are seeing an intermittent issue. _sendparent_ uses context updated by preceding assign action. Most of the time it is working fine, once in a while the updated context seems to be not available with _sendparent_ by the time it is invoked or that's what we are suspecting. If this is not possible, then we have to look elsewhere for why this is happening.
